### ISSUE TITLE
Layer defined at provider level triggers an error - seems it is trying to install it twice

### ISSUE THREAD

**Issue by bendubuisson** [DataDog/serverless-plugin-datadog#195]
## Expected Behavior
Add a layer at the provider level, it gets added to every function.

## Actual Behavior
SLS fails with error 

```
 Serverless Error ----------------------------------------
 
An error occurred: MyLambdaFunction - Resource handler returned message: "Two different versions of the same layer are not allowed to be referenced in the same function. arn:aws:lambda:ap-southeast-2:***:layer:test-python-requirements:12 and arn:aws:lambda:ap-southeast-2:***:layer:test-python-requirements:12 are versions of the same layer. (Service: Lambda, Status Code: 400, Request ID: xxxxxxx, Extended Request ID: null)" (RequestToken:xxxxxxx, HandlerErrorCode: InvalidRequest).
```
Seems suspiciously related to https://github.com/DataDog/serverless-plugin-datadog/issues/186 and https://github.com/DataDog/serverless-plugin-datadog/pull/188

Moving the layer to the function fixes the issue. Obviously not ideal.

## Steps to Reproduce the Problem

Have a layer specified at the provider level, none at the function level and the following sls config:

```
custom:
  prune: 
    automatic: true
    number: 3
  warmup: 
    default:
      enabled: true
  pythonRequirements:
    layer: true
  datadog:
    apiKey:MYKEY
    addExtension: true
    enableDDTracing: true
    enableDDLogs: true
    flushMetricsToLogs: true
    exclude:
      - warmUpPluginDefault
```
## Specifications

  - Serverless Framework version: 2.64.1
  - Datadog Serverless Plugin version: 3.1.2
  - Lambda function runtime (Python 3.7, Node 10, etc.): Python 3.9
  
Thank you !

**Comment by astuyve** [comment]
Fixed in 3.2.0

**Comment by bendubuisson** [comment]
Awesome work, much appreciated!
